### ISSUE TITLE
Attribut "hidden" für readme.md

### ISSUE THREAD

**Issue by joesch2** [nextcloud/text#977]
When I use the workspace above the file list, a file with the name "readme.md" is created in this file list. It would helpful, if this "system file" is given the attribute "hidden" and is normally not displayed (unless I activate the display of hidden files).

**Comment by juliushaertl** [comment]
I think that would make sense to have as an option, however we make it not-hidden by intention, so that the file is easily visible when synced to the desktop as well.

cc @nextcloud/designers

**Comment by skjnldsv** [comment]
I agree :)

**Comment by kesselb** [comment]
I'm against it. It's similar to translated readme.md (e.g. liesmich.md) and adds a bunch of complexity. Even if the behavior is something to opt in. Imagine you have a bunch of readme.md files and turn the hidden feature on. You actually need a migration feature otherwise people see the readme.md but the Rich workspace is empty.

**Comment by JPUnD** [comment]
You could add an option to read both and then the one that shows on the rich workspace is the non-hidden if both an hidden and non-hidden exist. 

This would be convenient for people who share their folder with others without any tech knowledge and just want to move around without having a readme polluting the file tree

**Comment by kesselb** [comment]
> You could add an option to read both and then the one that shows on the rich workspace is the non-hidden if both an hidden and non-hidden exist.

That works for 50%. The other 50% want the non-hidden if both exist. Thus we need another option to configure that behavior.

> others without any tech knowledge 

I think it's simple even for people without tech knowledge to understand the connection between rich workspace and `README.md`. If not they might ask their operator. It's not super hard to explain.

**Comment by JPUnD** [comment]
Yeah fair enough, having an option to change it's behavior would in fact be preferable. 

And I guess adding information about the `Readme.md` on the top level directory for warning.

**Comment by jancborchardt** [comment]
Agree with the concerns from @kesselb and partly @juliushaertl.

This can lead to confusion, especially when it comes to desktop sync where we have no way of directly showing the content of the readme, and then for regular people who don’t know about hidden files, there is no way to access this normally very present part of the folder.

A middle ground idea I thought about but don’t think is good anymore: When people rename the "Readme.md" to ".Readme.md" it could still work but be hidden. **We shouldn’t do that either since then the behavior changes even from folder to folder and gets even more confusing.** It would essentially be an interface feature that every individual person can switch on and off for everyone for every folder – total mess and goes against expectations.

TL;DR let’s keep it simple as it is? At least until we have a proper way of integrating this in desktop – which doesn’t look likely.

**Comment by flintstryker** [comment]
Just wanted to leave my 2 cents on this issue. I'm an admin that finds the readme.md file to be very annoying when it shows up in every folder simply because the top area was clicked by someone. Surely there is a simple way to remedy this. I honestly don't understand the concerns mentioned in the responses. 1. I don't think you have to migrate between hidden and non-hidden readme files; if necessary just include a little warning that the admin will need to do the cleanup themselves. It's better than not having the option to use hidden readme files.  2. I don't think you have to sync hidden readme files. I know this won't sit well with some of you, but really it's better than not having the option to use hidden readme files. We don't need a perfect solution, we just need options.

**Comment by dartcafe** [comment]
See here #1171. The automatically creation of a readme is bad rather bad UX, because users would not expect that to happen.

**Comment by kuenzelmedien** [comment]
As a workaround you can use the "Custom CSS" plugin and add:

To Hide Readme.md only for Public View (Share Links)
`#body-public tr[data-file="Readme.md"] {
display: none !important;
}`

To Hide Readme.md global
`tr[data-file="Readme.md"] {
display: none !important;
}`
